### PR TITLE
Follow Node importing semantics

### PR DESCRIPTION
**Pull request by kklin**

### CODE REVIEW

**Comment by kklin** [comment]
Btw, this importing a json file will likely be useful for configuration files. It's pretty neat.

File: stitch/get.go [hunk 1]
```diff
@@ -133,19 +135,95 @@ func (getter ImportGetter) checkSpec(file string, _ os.FileInfo, _ error) error
...
 
-func (getter ImportGetter) specContents(name string) (string, string, error) {
-	modulePath := filepath.Join(getter.Path, name+".js")
-	if _, err := util.AppFs.Stat(modulePath); os.IsNotExist(err) &&
+// Error thrown when there are no files module files that can be read from disk.
+var errNoLoadableFile = errors.New("no loadable file")
+
+// loadAsFile searches for and evaluates `imp`, `imp`.js`, and finally `imp`.json.
+// Once a loadable import file is found, it stops searching.
+func loadAsFile(vm *otto.Otto, imp string) (otto.Value, error) {
```
Comment (ejj) [inline stitch/get.go]: `imp`? Not sure what that means.  Perhaps there's a more descriptive variable name?

File: stitch/get.go [hunk 2]
```diff
@@ -133,19 +135,95 @@ func (getter ImportGetter) checkSpec(file string, _ os.FileInfo, _ error) error
...
+		if path := imp + suffix; isFile(path) {
+			spec, err := util.ReadFile(path)
+			if err != nil {
+				return otto.Value{}, err
+			}
+			return runSpec(vm, path, spec)
+		}
+	}
+
+	if path := imp + ".json"; isFile(path) {
```
Comment (ejj) [inline stitch/get.go]: Shouldn't block this commit, but we really need to document all this stuff soonish.  Could you file a trello card?  

Also, so what are the semantics of importing a json file out of curiousity?

File: stitch/get.go [hunk 3]
```diff
@@ -133,19 +135,95 @@ func (getter ImportGetter) checkSpec(file string, _ os.FileInfo, _ error) error
...
+		if err != nil {
+			return otto.Value{}, err
+		}
+		return vm.ToValue(unmarshalled)
+	}
+
+	return otto.Value{}, errNoLoadableFile
+}
+
+// loadAsDir searches for and evaluates `dir`/package.json, and `dir`/index.
```
Comment (ejj) [inline stitch/get.go]: index.json?

File: stitch/get.go [hunk 4]
```diff
@@ -133,19 +135,95 @@ func (getter ImportGetter) checkSpec(file string, _ os.FileInfo, _ error) error
...
+// Once a loadable import file is found, it stops searching.
+func loadAsDir(vm *otto.Otto, dir string) (otto.Value, error) {
+	if path := filepath.Join(dir, "package.json"); isFile(path) {
+		pkgIntf, err := unmarshalFile(path)
```
Comment (ejj) [inline stitch/get.go]: Intf?

**Comment by ejj** [comment]
Some minor nits, but I'm happy with this.  Needs to be rebased, but it's ready

File: stitch/get.go (hunk 3 above)
Comment (kklin) [inline stitch/get.go]: I meant that it loads the import `dir`/index as an import according to the file loading rules. I've changed the comment to be clearer.

File: stitch/get.go (hunk 2 above)
Comment (kklin) [inline stitch/get.go]: We have an existing card, but I moved it to up next. It'd be helpful for me if you outlined headings or something on general topics you'd like us to cover.

We unmarshal the JSON file into a javascript object.
```
// config.json
{
  "namespace": "kklin",
}

// foo.js
var config = require("./config.json");
console.log(config["namespace"]); // kklin
```

**Comment by kklin** [comment]
Addressed comments.

**Pull request merged**